metrics/core.go

```diff
@@ -1,119 +1,74 @@
+func (m Metrics) RecPutBackendRequest(status string, duration *time.Time, sizeInBytes float64) {
+	for _, me := range m.MetricEngines {
+		me.RecordPutBackendRequest(status, duration, sizeInBytes)
 	}
 }
-
-func NewConnectionMetrics(r metrics.Registry) *ConnectionMetrics {
-	return &ConnectionMetrics{
-		ActiveConnections:      metrics.GetOrRegisterCounter("connections.active_incoming", r),
-		ConnectionAcceptErrors: metrics.GetOrRegisterMeter("connections.accept_errors", r),
-		ConnectionCloseErrors:  metrics.GetOrRegisterMeter("connections.close_errors", r),
+func (m Metrics) RecGetBackendRequest(status string, duration *time.Time) {
+	for _, me := range m.MetricEngines {
+		me.RecordGetBackendRequest(status, duration)
 	}
 }
-
-type Metrics struct {
-	Registry        metrics.Registry
-	Puts            *MetricsEntry
-	Gets            *MetricsEntry
-	PutsBackend     *MetricsEntryByFormat
-	GetsBackend     *MetricsEntry
-	Connections     *ConnectionMetrics
-	ExtraTTLSeconds metrics.Histogram
+func (m Metrics) RecConnectionMetrics(status string) {
+	for _, me := range m.MetricEngines {
+		me.RecordConnectionMetrics(status)
+	}
 }
-
-// Export begins sending metrics to the configured database.
-// This method blocks indefinitely, so it should probably be run in a goroutine.
-func (m *Metrics) Export(cfg config.Metrics) {
-	switch cfg.Type {
-	case config.MetricsInflux:
-		logrus.Infof("Metrics will be exported to Influx with host=%s, db=%s, username=%s", cfg.Influx.Host, cfg.Influx.Database, cfg.Influx.Username)
-		influxdb.InfluxDB(
-			m.Registry,          // metrics registry
-			time.Second*10,      // interval
-			cfg.Influx.Host,     // the InfluxDB url
-			cfg.Influx.Database, // your InfluxDB database
-			cfg.Influx.Username, // your InfluxDB user
-			cfg.Influx.Password, // your InfluxDB password
-		)
-	case config.MetricsNone:
-		return
-	default:
-		logrus.Fatalf("Unrecognized config metrics.type: %s", cfg.Type)
+func (m Metrics) RecExtraTTLSeconds(value float64) {
+	for _, me := range m.MetricEngines {
+		me.RecordExtraTTLSeconds(value)
 	}
-	return
 }
 
-func CreateMetrics() *Metrics {
-	flushTime := time.Second * 10
-	r := metrics.NewPrefixedRegistry("prebidcache.")
-	m := &Metrics{
-		Registry:        r,
-		Puts:            NewMetricsEntry("puts.current_url", r),
-		Gets:            NewMetricsEntry("gets.current_url", r),
-		PutsBackend:     NewMetricsEntryBackendPuts("puts.backend", r),
-		GetsBackend:     NewMetricsEntry("gets.backend", r),
-		Connections:     NewConnectionMetrics(r),
-		ExtraTTLSeconds: metrics.GetOrRegisterHistogram("extra_ttl_seconds", r, metrics.NewUniformSample(5000)),
+func (m Metrics) Export(cfg config.Configuration) {
+	for _, me := range m.MetricEngines {
+		me.Export(cfg.Metrics)
 	}
+}
 
-	metrics.RegisterDebugGCStats(m.Registry)
-	metrics.RegisterRuntimeMemStats(m.Registry)
+/* Interface definition                */
+type CacheMetrics interface {
+	RecordPutRequest(status string, duration *time.Time)
```

Comment:
You use `Record` here and `Rec` in other methods. Consider using `Record` everywhere for consistency.

Reply:
`RecPutRequest` is a method of the `Metrics` object which will execute the `RecordPutRequest()` implementation of the `CacheMetrics` interface objects. If we name both the `Metrics` object method, and the `CacheMetrics` interface function the same (`RecordPutRequest()`) don't we risk it'll get confusing. (Even more confusing)